src/menu/stateful-container.js

```diff
@@ -286,6 +286,11 @@ export default class MenuStatefulContainer extends React.Component<
       event.preventDefault();
       onItemSelect({item: items[highlightedIndex], event});
     }
+    if (this.getItems().length >= this.state.highlightedIndex) {
```

Comment:
```suggestion
    if (items.length >= this.state.highlightedIndex) {
```

Reply:
Does this bug also come up when clicking with the mouse? Does it need to be handled somewhere else other than the `handleEnterKey` method?

Reply:
You are definitely right here, the safest way should be to put it in right after we update the menu value.